plugins/optimization-detective/tests/test-class-od-url-metrics-group-collection.php

```diff
@@ -199,6 +199,89 @@ public function data_provider_sample_size_and_breakpoints(): array {
 		);
 	}
 
+	/**
+	 * Test clear_cache().
+	 *
+	 * @covers ::clear_cache
+	 * @covers OD_URL_Metric_Group::clear_cache
+	 * @covers OD_URL_Metric::unset
+	 * @covers OD_Element::unset
+	 */
+	public function test_clear_cache(): void {
+		$collection      = new OD_URL_Metric_Group_Collection( array(), md5( '' ), array(), 1, DAY_IN_SECONDS );
+		$populated_value = array( 'foo' => true );
+		$group           = $collection->get_first_group();
+
+		// Get private members.
+		$collection_result_cache_reflection_property = new ReflectionProperty( OD_URL_Metric_Group_Collection::class, 'result_cache' );
+		$collection_result_cache_reflection_property->setAccessible( true );
+		$this->assertSame( array(), $collection_result_cache_reflection_property->getValue( $collection ) );
+		$group_result_cache_reflection_property = new ReflectionProperty( OD_URL_Metric_Group::class, 'result_cache' );
+		$group_result_cache_reflection_property->setAccessible( true );
+		$this->assertSame( array(), $group_result_cache_reflection_property->getValue( $group ) );
+
+		// Test clear_cache() on collection.
+		$collection_result_cache_reflection_property->setValue( $collection, $populated_value );
+		$collection->clear_cache();
+		$this->assertSame( array(), $collection_result_cache_reflection_property->getValue( $collection ) );
+
+		// Test that adding a URL metric to a collection clears the caches.
+		$collection_result_cache_reflection_property->setValue( $collection, $populated_value );
+		$group_result_cache_reflection_property->setValue( $group, $populated_value );
+
+		add_filter(
+			'od_url_metric_schema_root_additional_properties',
+			static function ( $schema ) {
+				$schema['new_prop_at_root'] = array(
+					'type' => 'string',
+				);
+				return $schema;
+			}
+		);
+		add_filter(
```

Comment:
all filters are removed after each test automatically, right? i often try to clean up any filters I add in tests out of habit which I believe is superfluous.

Reply:
Yes, filters are automatically reset after each test.

Reply:
https://github.com/WordPress/wordpress-develop/blob/e2b0bfdc06b2f652ff7b3865b496325b87521d26/tests/phpunit/includes/abstract-testcase.php#L202C10-L202C24